README.md

```diff
@@ -64,6 +64,7 @@ generate notifications over the webhook. So if you have a repository with little
 |--------------------|----------|----------|--------------------------------------------------------------------------|
 | `skip_download`    | No       | `true`   | Use with `get_params` in a `put` step to do nothing on the implicit get. |
 | `integration_tool` | No       | `rebase` | The integration tool to use, `merge` or `rebase`. Defaults to `merge`.   |
+| `git_depth`        | No       | `1`      | Shallow clone the repository using the `--depth` Git option              |
```

Comment:
I'm not sure I follow (I might be lacking some context). Can you explain a bit?

Reply:
I meant that we are setting depth only for the git pull, but the subsequent fetch of the PR does not limit the depth - but we can't really set the depth when fetching the PR since the build might be running for a commit which is not the tip of the PR branch. Make sense?🤔

Reply:
@itsdalmo I see it now, thanks. I've missed that part.

Are you sure about

> the build might be running for a commit which is not the tip of the PR branch

The relevant code seems to be

```go
cmd := g.command("git", "fetch", endpoint, fmt.Sprintf("pull/%s/head", strconv.Itoa(prNumber)))
```

which I think will always fetch the tip of the PR branch, doesn't it?

If that's the case, I would apply the depth setting there too (or maybe split the setting into `git_depth_base` and `git_depth`, but that sounds highly speculative)

Reply:
The Concourse job might be running for a commit which is not the tip of the PR branch in two cases; if someone pushes a commit after the job has been triggered, or the job is configured to trigger with `version: every` for the pull request resource.

> which I think will always fetch the tip of the PR branch, doesn't it?

I was unsure about all of this, so I did some testing with `depth`:

```bash
mkdir -p test-repository && cd test-repository && git init
git checkout -b master

# Pull
git pull git@github.com:itsdalmo/test-repository.git master --depth 3

# Fetch
git fetch git@github.com:itsdalmo/test-repository.git pull/4/head:my_second_pull --depth 6

# Merge
git merge a5114f6 --no-stat
```
Learnings:
- By default, `fetch` will pull in all the history for both the base and the PR. 
- We need to have enough `depth` to include the commit at which the two branches diverged in order to merge them (if not, git complains about history being unrelated).

With this in mind I agree that we should limit depth on `fetch` as well 👍

Reply:
Great, thank you for the analysis! I've just pushed the change to make the `fetch` step use `depth` as well